GeneratorInterface/Core/src/WeightHelper.cc

```diff
@@ -85,45 +78,45 @@ namespace gen {
     return "";
   }
 
-  void WeightHelper::updateScaleInfo(const ParsedWeight& weight, int index) {
-    auto& group = weightGroups_[index];
-    auto& scaleGroup = dynamic_cast<gen::ScaleWeightGroupInfo&>(group);
+  void WeightHelper::updateScaleInfo(gen::ScaleWeightGroupInfo& scaleGroup, const ParsedWeight& weight) {
     std::string muRText = searchAttributes("mur", weight);
     std::string muFText = searchAttributes("muf", weight);
-
-    if (muRText.empty() || muFText.empty()) {
+    std::string dynNumText = searchAttributes("dyn", weight);
+    float muR, muF;
+    try {
+      muR = std::stof(muRText);
+      muF = std::stof(muFText);
+    } catch (...) {
+      if (debug_)
+        std::cout << "Tried to convert (" << muR << ", " << muF << ") to a int" << std::endl;
       scaleGroup.setIsWellFormed(false);
       return;
+      /// do something
     }
 
-    try {
-      float muR = std::stof(muRText);
-      float muF = std::stof(muFText);
-      std::string dynNumText = searchAttributes("dyn", weight);
-      if (dynNumText.empty()) {
-        scaleGroup.setMuRMuFIndex(weight.index, weight.id, muR, muF);
-      } else {
-        std::string dynType = searchAttributes("dyn_name", weight);
+    if (dynNumText.empty()) {
+      scaleGroup.setMuRMuFIndex(weight.index, weight.id, muR, muF);
+    } else {
+      std::string dynType = searchAttributes("dyn_name", weight);
+      try {
         int dynNum = std::stoi(dynNumText);
-        scaleGroup.setMuRMuFIndex(weight.index, weight.id, muR, muF, dynNum, dynType);
+        scaleGroup.setDyn(weight.index, weight.id, muR, muF, dynNum, dynType);
+      } catch (...) {
+        std::cout << "Tried to convert (" << dynNumText << ")  a int" << std::endl;
+        scaleGroup.setIsWellFormed(false);
+        /// do something here
       }
-    } catch (std::invalid_argument& e) {
-      scaleGroup.setIsWellFormed(false);
     }
+
     if (scaleGroup.lhaid() == -1) {
       std::string lhaidText = searchAttributes("pdf", weight);
       try {
         scaleGroup.setLhaid(std::stoi(lhaidText));
-      } catch (std::invalid_argument& e) {
+      } catch (...) {
         scaleGroup.setLhaid(-2);
```

Comment:
To be honest, I don't remember exactly now, the blame doesn't give a lot of help. I think the intent was to distinguish the default value (-1) from an error in the LHAid text (ie pdf info in the xml but its not an integer). Considering I can't fully remember/find the intent of this code means it can be cleaned up... If the distinction between the two doesn't matter, we can just set it to -1 like the default value, else I can make some sort of enum to clarify this distinction.

Reply:
yeah it's confusing like this, lets revert and fix again if need be